docs/network.md

```diff
@@ -0,0 +1,220 @@
+# 4. Code flow - finding first peer to connect to.
+Each node maintains list of peers.
+
+New node will get an initial list of peers from:
+- Genesis config `network.boot_nodes`
+- Command line argument `--boot_nodes`
+
+Peer to connect to is chosen at random from list of known nodes by `PeerManagerActor::sample_random_peer` method.
+
+# 5. Edges & network - in code representation
+`P2P network` is represented by list of `peers`, where each `peer` is represented by structure `PeerId`,
+which is defined by `peer's` public key `PublicKey`.
+And a list of edges, where each edge is represented by the structure `Edge`.
+
+Both are defined below.
+
+# 5.1 PublicKey
+We use two types of public keys:
+- a 256 bit `ED25519` public key
+- a 512 bit `Secp256K1` public key
+
+Public keys are defined in `PublicKey` enum, which consists of those two variants.
+```rust
+pub struct ED25519PublicKey(pub [u8; 32]);
+pub struct Secp256K1PublicKey([u8; 64]);
+pub enum PublicKey {
+    ED25519(ED25519PublicKey),
+    SECP256K1(Secp256K1PublicKey),
+}
+```
+
+# 5.2 PeerId
+Each `peer` is uniquely defined by its `PublicKey`, and represented by `PeerId` struct.
+```rust
+pub struct PeerId(PublicKey);
+```
+
+# 5.3 Edge
+Each `edge` is represented by `Edge` struct. It's documented below:
+
+```rust
+pub struct Edge {
```

Comment:
it will be hard to keep this documentation in sync with the code. Maybe here just mention what is inside, and point to the file where it is implemented.

Reply:
Ok, I rewrote this section.